### PR TITLE
MMT-1719 Approver E-mails when a Proposal is Submitted

### PR DESCRIPTION
**Pull request by ryanmiller-1**
Updated how I was getting the approvers' URS to send the submission e-mails.  This method would tie the e-mails to whomever has the create permissions for approvers in SCIOPS.

### CODE REVIEW

File: app/controllers/proposal/collection_draft_proposals_controller.rb [hunk 1]
```diff
@@ -226,6 +230,44 @@ def submitter_from_resource
       { name: "#{proposal_urs_user['first_name']} #{proposal_urs_user['last_name']}", email: proposal_urs_user['email_address'] } unless proposal_urs_user.blank?
     end
 
+    # Get urs users and get emails from urs for approvers
```
Comment (ryanmiller-1) [inline app/controllers/proposal/collection_draft_proposals_controller.rb]: This function represents the bulk of the changes between this PR and the last one.

File: app/controllers/proposal/collection_draft_proposals_controller.rb [hunk 2]
```diff
@@ -226,6 +230,44 @@ def submitter_from_resource
...
+    # Returns array of hashes for each user
+    def get_approver_emails
```
Comment (charleshuang80) [inline app/controllers/proposal/collection_draft_proposals_controller.rb]: This is a nice idea, getting the emails from the ACLs. We should make sure we are getting all the right emails -- in SIT and UAT we check 'SCIOPSTEST' for the Draft MMT ACLs as well.

File: spec/support/approved_proposals_helpers.rb [hunk 3]
```diff
@@ -78,5 +78,20 @@ def mock_urs_get_users(count: 1, users: [])
       urs_response = cmr_success_response(urs_response_body.to_json)
       allow_any_instance_of(Cmr::UrsClient).to receive(:get_urs_users).and_return(urs_response)
     end
+
+    # Fake getting ACLs from CMR, then getting groups from CMR, then getting group members from CMR
+    def mock_get_approver_emails
```
Comment (charleshuang80) [inline spec/support/approved_proposals_helpers.rb]: Mocking most of this is fine for some of the tests, but we should have at least one instance where we are running through as many of the endpoints we can talk to in test (i.e. CMR) as possible.

File: app/controllers/proposal/collection_draft_proposals_controller.rb [hunk 4]
```diff
@@ -226,6 +230,54 @@ def submitter_from_resource
...
+      provider_targets = if Rails.env.production?
```
Comment (ryanmiller-1) [inline app/controllers/proposal/collection_draft_proposals_controller.rb]: Wrote MMT-2130 to move these to <environment>.rb files (needs to be done in other locations as well).

**Pull request merged**